### ISSUE TITLE
The upload of zip archive with folders fails

### ISSUE THREAD

**Issue by pierantoniomerlino** [eclipse/kura#3107]
**Describe the bug**
The CommandService can be used to upload compressed files to the gateway. If the zip file contains directories, the procedure fails with the following error:

```
2020-10-16T10:33:57,518 [pool-25-thread-1] ERROR o.e.k.c.a.c.CommandCloudApp - Error unzipping command zip bytes
java.io.FileNotFoundException: /tmp/hcitrace/hcitrace.txt (Not a directory)
```

**To Reproduce**
1. Create a zip file with a folder inside
2. Upload to the gateway with Device->Command tab
3. Check that there is an error in `kura.log` and in the /tmp folder there is a file with the same name of the folder in the zip archive.

**Expected behavior**
No errors on the logs and the content of the zip file should be decompressed on the /tmp folder

**Comment by MMaiero** [comment]
@LeoNerdoG can we define some test cases for this?

**Comment by LeoNerdoG** [comment]
Hey @MMaiero, yes we can! 😄  I will put it on my list and discuss together with testing team and try to get to it asap.

**Comment by MMaiero** [comment]
We decided to leave this open for further analysis

**Comment by MMaiero** [comment]
Closing as implemented fully now after #3116 for Kura 5.0.0 and #3117 for 4.1.2
